wagtail/admin/checks.py

```diff
@@ -237,3 +237,41 @@ def file_overwrite_check(app_configs, **kwargs):
         )
 
     return errors
+
+
+@register("datetime_format")
+def datetime_format_check(app_configs, **kwargs):
+    from django.conf import settings
+    from django.utils import formats, translation
+
+    errors = []
```

Comment:
```suggestion
    errors = []

    if not settings.USE_L10N:
        return errors
```

To avoid unnecessary checks, lets' return early if localisation is not in use.